actions/approve_csr_handler.go

```diff
@@ -54,21 +61,27 @@ func (h *approveCSRHandler) Handle(ctx context.Context, data interface{}) error
 	})
 }
 
-func (h *approveCSRHandler) handle(ctx context.Context, log logrus.FieldLogger, cert *csr.Certificate) error {
-	if cert.Approved() {
-		log.Debug("initial csr is already approved")
-		return nil
-	}
-
+func (h *approveCSRHandler) handle(ctx context.Context, log logrus.FieldLogger, cert *csr.Certificate) (reterr error) {
 	// Since this new csr may be denied we need to delete it.
 	log.Debug("deleting old csr")
 	if err := csr.DeleteCertificate(ctx, h.clientset, cert); err != nil {
 		return fmt.Errorf("deleting csr: %w", err)
 	}
 
+	var newCert *csr.Certificate
+	var err error
+
+	defer func() {
+		if errors.Is(reterr, context.Canceled) && newCert == nil {
+			// Rollback csr deletion if cluster-controller is restarted.
```

Comment:
Can you explain here more why we are requesting a certificate once more?

Reply:
if cluster controller gets restarted in the middle of approval, we will lose csr forever

Reply:
tested this again and it looks like kubelet keeps requesting csr if it gets deleted so we don't need this rollback